pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -246,6 +264,49 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	return nil
 }
 
+// cleanupErroredSnapshots cleans any volumesnapshot CRs that are in Error state. It doesn't remove them
+// from the volumesnapshotschedule items list though.
+func (s *SnapshotScheduleController) cleanupErroredSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) (err error) {
+	// Filter out the errored out snapshots that are older than the cleanup cutoff period.
+	for policy, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
+		for idx, snapshot := range policyVolumeSnapshot {
+			if snapshot.Status == snapv1.VolumeSnapshotConditionError && time.Since(snapshot.CreationTimestamp.Time) > errorSnapshotCleanupCutoffTime {
```

Comment:
Let's add one more condition here. `!snapshot.Deleted` . Basically ignore thosa snapshots statuses for which `Deleted` has already been set.

Reply:
Done